tests/interfaces/records-read.spec.ts

```diff
@@ -22,6 +23,93 @@ describe('RecordsRead', () => {
 
       expect(recordsRead.message.descriptor.messageTimestamp).to.equal(currentTime);
     });
+
+    it('should require `recordId` when `protocol` and `protocolPath` are not set', async () => {
+      const alice = await TestDataGenerator.generatePersona();
+      const currentTime = getCurrentTimeInHighPrecision();
+      const readPromise = RecordsRead.create({
+        authorizationSignatureInput : Jws.createSignatureInput(alice),
+        date                        : currentTime
+      });
+
+      await expect(readPromise).to.be.rejectedWith('must have required property \'recordId\'');
+    });
+
+    it('should require `recordId` when `protocolPath` is set and `protocol` is not set', async () => {
+      const alice = await TestDataGenerator.generatePersona();
+      const currentTime = getCurrentTimeInHighPrecision();
+      const readPromise = RecordsRead.create({
+        protocolPath                : 'some/path',
+        authorizationSignatureInput : Jws.createSignatureInput(alice),
+        date                        : currentTime
+      });
+
+      await expect(readPromise).to.be.rejectedWith('must have required property \'recordId\'');
+    });
+
+    it('should require `recordId` when `protocol` is set and `protocolPath` is not set', async () => {
```

Comment:
Should we make it such that `recordId` and `protocol + protocolPath` are mutually exclusive?

Reply:
Yup, that's actually how the behavior is expected, but these tests were structured/worded poorly.

I've gone ahead and restructured the tests to better represent the intention, but this brought out that the error returned by the JSON Schema validator always asks for a `recordId`, which isn't really descriptive enough.

I was thinking about adding a specific DwnError here for create and failing fast before it even gets down to the Schema validation. This way the error can be more descriptive similar to the one thrown when receiving a bad RecordsRead during handle.

I'll still keep the rules that exist within the JSON Schema, but would just throw a better error before it gets there.

Reply:
I ended up doing this and added a DwnError: `RecordsReadMissingCreateProperties`